depthai_ros_driver/src/dai_nodes/sensors/rgb.cpp

```diff
@@ -0,0 +1,114 @@
+#include "depthai_ros_driver/dai_nodes/sensors/rgb.hpp"
+
+#include "cv_bridge/cv_bridge.h"
+#include "depthai_bridge/ImageConverter.hpp"
+#include "image_transport/camera_publisher.hpp"
+#include "image_transport/image_transport.hpp"
+
+namespace depthai_ros_driver {
+namespace dai_nodes {
+RGB::RGB(const std::string& daiNodeName,
+         rclcpp::Node* node,
+         std::shared_ptr<dai::Pipeline> pipeline,
+         dai::CameraBoardSocket socket = dai::CameraBoardSocket::RGB,
+         sensor_helpers::ImageSensor sensor = {"IMX378", {"12mp", "4k"}, true})
+    : BaseNode(daiNodeName, node, pipeline) {
+    RCLCPP_INFO(node->get_logger(), "Creating node %s", daiNodeName.c_str());
+    setNames();
+    colorCamNode = pipeline->create<dai::node::ColorCamera>();
+    ph = std::make_unique<param_handlers::RGBParamHandler>(daiNodeName);
+    ph->declareParams(node, colorCamNode, socket, sensor);
+    setXinXout(pipeline);
+    RCLCPP_INFO(node->get_logger(), "Node %s created", daiNodeName.c_str());
+};
+void RGB::setNames() {
+    colorQName = getName() + "_color";
+    previewQName = getName() + "_preview";
+    controlQName = getName() + "_control";
+}
+
+void RGB::setXinXout(std::shared_ptr<dai::Pipeline> pipeline) {
+    if(ph->getParam<bool>(getROSNode(), "i_publish_topic")) {
+        xoutColor = pipeline->create<dai::node::XLinkOut>();
+        xoutColor->setStreamName(colorQName);
+        colorCamNode->video.link(xoutColor->input);
```

Comment:
`colorCamNode->isp.link` might be better here ?
https://docs.luxonis.com/projects/api/en/latest/components/nodes/color_camera/